### ISSUE TITLE
EditButton will not re-render after custom props changes (className)

### ISSUE THREAD

**Issue by Slavvkko** [marmelab/react-admin#3910]
**What you were expecting:**
Change custom props in `EditButton`  (e.g. `className`) trigger button re-render.

**What happened instead:**
`EditButton` wont re-render.

**Steps to reproduce:**
In root component:
```
const useStyles = makeStyles({
  editButton: {
    display: "none"
  }
});

function MyComponent() {
  const classes = useStyles();

  return (
      <EditButton
        className={classes.editButton}
        basePath={basePath}
        record={record}
      />
  )
}
```

After root component re-renders, `useStyles` will return new name for `editButton` and `EditButton` will omit this change due to shouldUpdate internal.

**Related code:**
From source of `EditButton`:

```
const enhance = shouldUpdate(
    (props, nextProps) =>
        props.translate !== nextProps.translate ||
        (props.record &&
            nextProps.record &&
            props.record.id !== nextProps.record.id) ||
        props.basePath !== nextProps.basePath ||
        (props.record == null && nextProps.record != null)
);

export default enhance(EditButton);
```

**Environment**

* React-admin version: `3.0.0-beta.3`
* Last version that did not exhibit the issue (if applicable):
* React version: `16.10.2`
* Browser: `Version 78.0.3904.70 (Official Build) Arch Linux (64-bit)`
* Stack trace (in case of a JS error): `not need`

**Comment by Kmaschta** [comment]
Thank you for opening this issue. In order to confirm it, we'll have to reproduce it.
As explained in the bug report template, please fork the following CodeSandbox and repeat your issue on it:

https://codesandbox.io/s/github/marmelab/react-admin/tree/next/examples/simple

This is the simplest way to confirm a bug is related to the React Admin codebase.

**Comment by fzaninotto** [comment]
Fixed by #3929
